### DISCUSSION TITLE
Compare the performance of if and match inside the rsx macro!

### DISCUSSION THREAD

**Discussion by RustGrow** [DioxusLabs/dioxus#3101] [Q&A] [answered]
Does using an rsx! macro inside another rsx! macro affect performance? 
With match:
```
rsx!{
     match new_map.get("first-name") {
        Some(first_name) => rsx!{
                    div { "First name: {first_name}" }
                },
        None => rsx!{
            div { "First name: empty" }
        },
    },
}
```
With if:
```
rsx!{
  if new_map.contains_key("first-name") {
      div { r#"First name: {new_map["first-name"]}"# }
  } else {
      div { "First name: empty" }
  }
}
```
What is the best approach to take?

**Comment by ealmloff** [comment, marked as answer]
If statements and match statements inside rsx just turns into rust's if statements and match statements. The performance difference will be the same. I would expect the first version (match) to be faster because you are getting the data from the hashmap at the same time you are checking if it exists, but the difference will likely be negligible either way
